### DISCUSSION TITLE
Releasing a tree-sitter grammar

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1332] [Q&A] [answered]
Possibly related: https://github.com/tree-sitter/tree-sitter/issues/1223

I am now at the point where [my grammar](https://github.com/tlaplus-community/tree-sitter-tlaplus) is good enough to be released and used by others. Where do people release their grammars? I guess the answer is probably wherever there exist compatible tree-sitter bindings packages? NPM and Rust Crates seem the most prominent, although Python might also be a good idea for me because a fair number of TLA+ meta-tools are created with Python. Does anybody have good examples of automated release pipelines they've set up for their grammars?

**Comment by razzeee** [comment]
NPM publish https://github.com/elm-tooling/tree-sitter-elm/blob/main/.github/workflows/npm-publish.yml 
(the github package manager usually errors out, it worked in the beginning, so far I didn't take time to fix it)

Rust publish https://github.com/elm-tooling/tree-sitter-elm/blob/main/.github/workflows/publish.yml

Binaries that get attached to your release 
https://github.com/elm-tooling/tree-sitter-elm/blob/main/.github/workflows/publish-prebuild.yml
See https://github.com/elm-tooling/tree-sitter-elm/releases


Here's how I release https://github.com/elm-tooling/tree-sitter-elm/blob/main/HOW_TO_RELEASE.md

**Reply by ahelwer** [reply]
Thank you, very helpful!

**Comment by ObserverOfTime** [comment, marked as answer]
https://github.com/tree-sitter/workflows
